cmake/SetupHDF5.cmake

```diff
@@ -0,0 +1,28 @@
+include(${CMAKE_CURRENT_LIST_DIR}/Util.cmake)
+
+function(setup_hdf5)
+    if(NOHDF5)
+        target_compile_definitions(${TARGET_NAME} PRIVATE NHDF5RH=1)
```

Comment:
take a function argument rather than hardcoding it to the global variable `TARGET_NAME`

Reply:
Done.